### DISCUSSION TITLE
Confused about the `hostname` option

### DISCUSSION THREAD

**Discussion by J0F3** [keycloak/keycloak#32864] [Q&A] [answered]
When I understand the documentation (https://www.keycloak.org/server/hostname) correctly the hostname option is the one which should be used with the hostname v2 configuration. But with that keycloak give the following warning on every startup:

```
WARN  [org.keycloak.quarkus.runtime.cli.Picocli] (main) The following used options or option values are DEPRECATED and will be removed or their behaviour changed in a future release:
        - hostname
```

So, I am bit confused how to configure the hostname correctly now.

Thanks!

**Comment by shawkins** [comment]
If you are on KC 25 this should mean that you also are specifying the features option with hostname:v1. If you don't explicitly enable hostname:v1, hostname:v2 will be enabled by default.

**Reply by J0F3** [reply]
That is what confused me. I am on KC 25 and I did not enable hostname:v1 explicitly, at least as far as I know. Can I see somewhere which version is used?

I just set the following setting through the keycloak.conf:

- `http-enabled=true` because HTTPS Offloading is done by Azure Container App Service.
- proxy-headers=xforwarded
- hostname="FQDN of the container app" e.g. https://keycloak.something-445b3fa2e.switzerlandnorth.azurecontainerapps.io/

So, I do not know how I get rid of the warning.

**Reply by J0F3** [reply]
I am seeing in the admin us also that the feature `HOSTNAME_V1` is even disabled (and `HOSTNAME_V2` enabled). So, I assume I can ignore the warning then?

**Reply by vmuzikar** [reply, marked as answer]
This is possibly the same issue as https://github.com/keycloak/keycloak/issues/30380. @J0F3 Can you confirm you use `--optimized`?

**Reply by J0F3** [reply]
Yes, that seems to be it and yes, I use `--optimized` Thanks!
